[CodeSec AG] Update Monitor and Fix Issues [RLP-92083] (#946)
PaloAltoNetworks/prisma-cloud-docs: code-security/admin_guide/scan-monitor/monitor-fix-issues-in-scan/monitor-fix-issues-in-scan.adoc

*Projects* provides a consolidated view of scan results across your repositories and CI/CD pipelines wherein categorization of issues are at resource or policy level giving you a better understanding on your security posture.

* *VCS main branch scans*: Periodic scans performed on all main branches across repositories.
* *VCS Pull Requests*: Event driven scans using Enforcement parameters are run on all open Pull Requests (PR) for your integrated repositories.
* *CLI and CI/CD runs*: Event driven scans performed on runs as configured by the you using the Enforcement parameters.

All three types of scans are run on code categories of IaC misconfiguration, Vulnerabilities, Secrets, Licenses and Build Integrity.

image::proj.png[width=800]

[cols="1,2", options="header"]
|===

| Projects Code Category View
| Description

|Overview
|A summary view of all your scan results across all code categories. The Overview lists errors by prioritizing build issues across the integrations.

|IaC Misconfiguration

> |The view lists security issues after scanning integrated repositories, wherein the default branch of a repository lists all code security violations in the resource block with code tags, resource dependencies and resource history. In this view, you can also identify Drift and trigger Manual Scans for your repository across any code categories view on the console.

Reviewer: again - Manual Scans is not relevant for.a specific code category

Author: Fixed.

|Build Integrity
|Misconfigured repositories and CI/CD pipelines can lead to Supply Chain attacks such as data exfiltration and code injection. Use this view to spot VCS and CI/CD misconfigurations.

|VCS Pull Requests
| After configuring Enforcement, view issues for scans run on all open Pull Requests (PR) for your integrated repositories. On the Prisma Cloud console, you can access the scan result from the console or choose to access the VCS console and view the specific commit within PR for a manual fix.

|CI/CD Runs
|View issues for scans performing on runs of your integrated CI/CD pipelines.

|===

You can also find relevant non-conformant scan results using filters or search using keywords or tags (if they are already a part of the code).
For each misconfiguration on *Projects*, you can either Suppress the issue, Fix it from the Prisma Cloud console, or access the repository in the Version Control System (VCS) to perform a manual fix.
Currently, this view of *Projects* is accessible on *Code Security > Projects > Enhanced*.

image::proj-1.png[width=800]

=== Resource Blocks

All issues are viewable as a resource on *Projects* across code category views.

* *Resource Block*
+

> After periodic scans on resources, Prisma Cloud generates contextualized scanned results of each resource as a resource block. Scan results are are vulnerabilities in the code or code errors found within the resource.

Reviewer: double "are"

Author: Fixed.

+
3. *Additional Information*: Displays columns of the information regarding the issue.
+
* *Policy*: Displays the severity level of non-conformant policy in the code.
+
* *First Detected*: The timestamp of the issue when found.

=== Resource Explorer and Fix Cart

In helping you make educated decision, Prisma Cloud provides you with granular information on each issue within Resource Explorer. Later each of the issues are remediated on Fix Cart.

==== Resource Explorer

The information on Resource Explorer enables you to make an educated decision on the security violation and understand if the violation has any connection as a dependency on other resources within the repository while exploring the change log of the resource.
You can view this contextualized information across four tabs.

* *Details*: Helps you understand the connection between resources while enabling you to make informed decisions if the connection is at risk or if it is necessary.
+
image::proj-7.png[width=400]

* *Errors*: Enables you to review security issues across all resource types with the package severity threshold and utilize the information to either fix, suppress or manually add a fix to the issue.

> +

Reviewer: it called now Issues and it show all issues across multiple statuses (suppressed/passed/etc)

Author: On app-stage stage when I checked last it was still Errors. And currently the site is not stable. I will hide the image for now and fix the wording.

Author: Fixed the wording.

Author: Fixed the wording.

Status for each scanned repository is created based on the non-conformance to a policy. The repository status can be further filtered as Errors, Suppressed and Passed.

image::proj-11.png[width=800]

[cols="1,2", options="header"]
|===

|Status
|Description

|Error
|A resource appears with an error status when it is non-conformant to a policy.

|Passed
|A resource that has conformant policies or may have a history of fixed errors.

|Suppressed
|A resource previously appeared with a non-conformant policy but is suppressed with a Suppress action. To suppress a non-conformant policy in a resource is when you absolve the scanned result with a definitive explanation indicating the non-conformance to be not problematic.

|Fix Pending
|A fix awaiting a PR merge in your VCS console.

|===

Your scanned resources appear on *Code Security > Projects* with an active Error filter by default. You can choose to add more filters or remove the Error filter.

[#severities-]
===== Severities

A Severities indicates an impact on a non-conformant resource in your repository. Resources can be filtered as Critical,High, Medium, Low and Informational in severity.

image::proj-12.png[width=800]

> 

Reviewer: There are more filters that should be documented - see here https://docs.bridgecrew.io/docs/projects_january23#additional-filters

Author: Added the additional filters.

|After configuring Enforcement, view issues for scans performing on runs of your integrated repositories. Using the open PR, you can access the specific run performed on the repository.

|===

You can also find relevant non-conformant scan results using filters or search using keywords or tags (if they are already a part of the code).
For each misconfiguration on *Projects*, you can either Suppress the issue, Fix it from the Prisma Cloud console, or access the repository in the Version Control System (VCS) to perform a manual fix.
Currently, this view of *Projects* is accessible on *Code Security > Projects > Enhanced*.

image::proj-1.png[width=800]

=== Resource and Policy Blocks

All issues are viewable as a resource or policy block on *Projects* across code category views.

* *Resource Block*
+
After periodic scans on resources, Prisma Cloud generates contextualized scanned results of each resource as a resource block. Scan results are issues that are vulnerabilities or code errors found within the resource.
Each resource block displays only five issues by default. *Show More* in each resource block helps you display more issues within the resource.

* *Policy Block*

> +

Author: Hidding this section as this is GA a few weeks from now. @GiladMark

Author: Hidding this section as this is GA a few weeks from now. @GiladMark

== Monitor and Fix Issues in Your Scans

toc::[]

> Prisma Cloud performs periodic scans on each integrated repository of the Version Control System (VCS) and CI/CD to find infrastructure misconfigurations.

Reviewer 2: ```suggestion
Prisma Cloud performs periodic scans on each integrated repository of onboarded Version Control Systems (VCS) and CI/CD pipelines to find infrastructure misconfigurations, open source vulnerabilities, license compliance violations, and exposed secrets.
```

We may need to highlight that VCS scans are periodic, CI/CD are on a trigger event

Author: Added @try-panwiac.

> *Projects* provides a consolidated view of scan results across your repositories, CI/CD and packages wherein categorization of issues are at resource or policy level giving you a better understanding on your security posture.

Reviewer 2: ```suggestion
*Projects* provides a consolidated view of scan results across your repositories and CI/CD pipelines wherein categorization of issues are at resource or policy level giving you a better understanding on your security posture.
```

Author: Added @try-panwiac.

Prisma Cloud performs three types of scans on your integrations.

* *VCS main branch scans*: Periodic scans performed on all main branches across repositories.

> * *VCS Pull Requests*: Periodic scans using Enforcement parameters are run on all open Pull Requests (PR) for your integrated repositories.

Reviewer 2: Not Periodic
```suggestion
* *VCS Pull Requests*: Event driven scans using Enforcement parameters are run on all open Pull Requests (PR) for your integrated repositories.
```

> * *CLI and CI/CD runs*: The Enforcement parameters with periodic scans are performed on all runs for your integrated repositories.

Reviewer 2: ```suggestion
* *CLI and CI/CD runs*: Event driven scans performed on runs as configured by the user using the Enforcement parameters.
```

All three types of scans are run on code categories of IaC misconfiguration, Vulnerabilities, Secrets, Licenses and Build Integrity.

image::proj.png[width=800]

[cols="1,2", options="header"]
|===

| Projects Code Category View
| Description

|Overview
|A summary view of all your scan results across all code categories. The Overview lists errors by prioritizing build issues across the integrations.

|IaC Misconfiguration

> |The view lists security issues after scanning integrated repositories, wherein the master branch of a repository lists all code security violations in the resource block with code tags, resource dependencies and resource history. In this view, you can also identify Drift and trigger Manual Scans for your repository.

Reviewer: let's call it "default" branch and not "master"

Reviewer: trigger manual scans is across all views - this is an action from the upper menu

|Vulnerabilities

> |The error list in Vulnerabilities are dependency errors that, after scanning packages and container images, are issues occurring as a part of your Supply Chain Security.

Reviewer 2: ```suggestion
|The error list in Vulnerabilities are open source dependency errors found in scanned open source packages and container image dependencies.
```

|Secrets

> |A Secret is a programmatic access key that provides systems access to information, services, or assets. Secrets like API keys, encryption keys, OAuth tokens, certificates, PEM files, passwords, and passphrases are often explicitly stored in local or feature branches before being pushed to a main branch. In this view, see issues identified in your source code with remediation to perform a Manual Fix or Suppress the issue.

Reviewer: we should mention this is shown in a file level

|Licenses

> |The usage of open source software licenses composes most of the source codes. At the same time, there might be an oversight regarding the permission and rights to use or repurpose an open source code. In this view, see the listing of licensing issues with remediation solutions through periodic scans.

Reviewer 2: ```suggestion
|Most open source software includes a license governing its use. License scanning identifies packages that violate open source usage policies. In this view, see the listing of licensing issues through periodic scans.
```

Reviewer 3: Good edits @try-panwiac . Thank you!

|Build Integrity

> |Repository misconfigurations occur when multiple owners for repositories exist on platforms like GitHub and GitLab. Through periodic scans, this view highlights the origin of another repository owner and prevents a chance Supply Chain attack.

Reviewer 2: ```suggestion
|Misconfigured repositories and CI/CD pipelines can lead to Supply Chain attacks such as data exfiltration and code injection. Use this view to spot VCS and CI/CD misconfigurations.
```

|VCS Pull Requests
| After configuring Enforcement, view issues for scans run on all open Pull Requests (PR) for your integrated repositories. On the Prisma Cloud console, you can access the scan result from the console or choose to access the VCS console and view the specific commit within PR for a manual fix.

|CI/CD Runs

> |After configuring Enforcement, view issues for scans performing on runs of your integrated repositories. Using the open PR, you can access the specific run performed on the repository.

Reviewer 2: ```suggestion
|View issues for scans performing on runs of your integrated CI/CD pipelines.
```

|===

You can also find relevant non-conformant scan results using filters or search using keywords or tags (if they are already a part of the code).
For each misconfiguration on *Projects*, you can either Suppress the issue, Fix it from the Prisma Cloud console, or access the repository in the Version Control System (VCS) to perform a manual fix.
Currently, this view of *Projects* is accessible on *Code Security > Projects > Enhanced*.

image::proj-1.png[width=800]

=== Resource and Policy Blocks

> All issues are viewable as a resource or policy block on *Projects* across code category views.

Reviewer: policy block will be available by EOQ3

* *Resource Block*
+

> After periodic scans on resources, Prisma Cloud generates contextualized scanned results of each resource as a resource block. Scan results are issues that are vulnerabilities or code errors found within the resource.

Reviewer: This is relevant also for CI/CD and PR scans

+
image::proj-3.png[width=800]
+
1. *Package Name and Path*: Displays the package name and it's code path.
+
2. *Total number of Issues*: Displays the total number of issues identified in the package.
+
3. *Additional Information*: Displays columns of the information regarding the issue.
+
* *CVE*: Displays the CVE name and the severity level of the violation.
+
* *Package*: View the violated package while identifying if the package is a `Root` or a dependent `package`. If the CVE exposed by a dependent package, you can see the name of the dependent package.
+
* *Root fix version*: Displays the recommended fix version for the root package to update.
+
* *CVSS*: Displays the Common Vulnerability Scoring System (CVSS).
+
* *Risk Factors*: Displays the risk factor of the CVE using Prisma Cloud defined values. The values range is Has Fix, Attach Complexity, DoS, Attack Vector, and Remote Execution.
+
* *First Detected*: The timestamp of the issue when found.

* *Secrets Resource Block*
+
As a secrets issue, there is an extensive information in the resource block.
+
image::proj-4.png[width=800]
+

> 1. *Secret Name and Path*: Displays the repository name and it's code path.

Reviewer: the "resource" here is a file so all secrets detected are in a file level and not repo

+
2. *Total number of Issues*: Displays the total number of issues identified in the repository.
+
3. *Additional Information*: Displays columns of the information regarding the issue.
+
* *Secret type*: Displays the severity level of the exposed secret in the code.
+
* *Risk Factors*: For Secrets there are three types of risk factors.
+

> ** IsPublic: Identifies if the repository storing the secret is publicly accessible or private.

Reviewer: Changed in the UI - Private repository/Public repository

+
image::proj-6.png[width=800]
+
1. *Branch Name and Path*: Displays the branch name and it's code path.
+
2. *Total number of Issues*: Displays the total number of issues identified in the repository.
+
3. *Additional Information*: Displays columns of the information regarding the issue.
+
* *Policy*: Displays the severity level of non-conformant policy in the code.
+
* *First Detected*: The timestamp of the issue when found.

=== Resource Explorer and Fix Cart

In helping you make educated decision, Prisma Cloud provides you with granular information on each issue within Resource Explorer. Later each of the issues are remediated on Fix Cart.

==== Resource Explorer

The information on Resource Explorer enables you to make an educated decision on the security violation and understand if the violation has any connection as a dependency on other resources within the repository while exploring the change log of the resource.
You can view this contextualized information across four tabs.

* *Details*: Helps you understand the connection between resources while enabling you to make informed decisions if the connection is at risk or if it is necessary.
+
image::proj-7.png[width=400]

> * *Errors*: Enables you to review security violations with the package severity threshold and utilize the information to either suppress or prioritize it.

Reviewer: now it's Issues and it shows all statuses and not only Errors across all resource types

+
image::proj-8.png[width=600]

* *History*: Explore detailed information about a resource, including suppression, change logs and fixes.
+
image::proj-9.png[width=600]

* *Traceability*: Explore and monitor connections between build-time and runtime resources.
//+
//image::[width=600]
+
The support for History and Traceability is currently only IaC resources, and the support for Errors is currently only available for packages.

> ==== Fix Cart

Reviewer: UI is now changed - please update it on Tuesday before we release it

A Fix Cart displays the issues you choose to fix before creating a Pull Request.

image::proj-10.png[width=600]

See xref:fix-issues-in-a-scan-result.adoc[Fix Issues in Scan] to know more on how to add issues to a fix cart.

[#filter-scan-results]
==== Filter Scan Results

Prisma Cloud enables you to filter your scan results across all code categories. You can filter your scan results across five default filters.

* <<repositories-,Repositories>>
* <<branch-,Branch>>
* <<code-categories,Code Categories>>
* <<issue-status,Issue Status>>
* <<severities-,Severities>>

[#repositories-]
===== Repositories

A list of integrated repositories.

image::proj-14.png[width=800]

[#branch-]
===== Branch

> A list of the supported branches of a VCS branch scan. Currently, the repository’s default branch is selected by default and cannot be configured

Reviewer: this is relevant only for the first six default views(non- PR/CI-CD runs)

[#code-categories]
===== Code Categories

A Category filters resources according to Build Integrity, Compute, Drift, General, IAM, Kubernetes, Licenses, Monitoring, Networking, Public, Secrets, Storage, and Vulnerabilities.
During the time of repositories integration on Prisma Cloud Code Security, your defined Categories associated with the repositories also help with filters.

image::proj-13.png[width=600]

[#issue-status]
===== Issue Status

Status for each scanned repository is created based on the non-conformance to a policy. The repository status can be further filtered as Errors, Suppressed and Passed.

image::proj-11.png[width=800]

[cols="1,2", options="header"]
|===

|Status
|Description

|Error
|A resource appears with an error status when it is non-conformant to a policy.

|Passed
|A resource that has conformant policies or may have a history of fixed errors.

|Suppressed
|A resource previously appeared with a non-conformant policy but is suppressed with a Suppress action. To suppress a non-conformant policy in a resource is when you absolve the scanned result with a definitive explanation indicating the non-conformance to be not problematic.

|Fix Pending

> |An issue awaiting remediation fix.

Reviewer: *fix.. as part of a fix pull request that is not yet to merged
